### PR TITLE
Updated API with stepper controller interface.

### PR DESCRIPTION
**Pull request by jazima**

### CODE REVIEW

File: api.py [hunk 1]
```diff
@@ -121,18 +129,46 @@ def move_coarse_focus(degrees: float, err_tol: float = 0.01) -> None:
...
+    else:
+        direction_packet = 0
+
+    steps_full = int(abs(degrees)//step_degrees) #round down to integer
+
+    while steps_full > 255:
+        steps_full -= 255
+        ser.write(motor_axis)
+        ser.write(direction_packet.to_bytes(1, byteorder = "big"))
+        ser.write(int(255).to_bytes(1, byteorder = "big"))
```
Comment (HPrickettMorgan) [inline api.py:169]: Would you prefer if we just used a big enough int (int32? int64?) for the protocol that didn't need to be split? If this works it's probably worth not messing with it though.

File: api.py [hunk 2]
```diff
@@ -32,14 +38,16 @@ def take_image() -> OpenCVImage:
 
 def stepper_controller_init() -> None:
     "Initializes the serial connection to the stepper controller."
-    raise NotImplementedError()
+    ser.open()
+    ser.timeout = 10
+    return
```
Comment (HPrickettMorgan) [inline api.py]: This return is unnecessary

**Comment by HPrickettMorgan** [comment]
Besides the extremely minor comments, this looks good to me. Let's merge after testing on the actual board/steppers.

File: stepper_controller.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+number of steps
+"""
+print(1)
+motor_axes = {
+    "x": int(0).to_bytes(1, "big"),
+    "y": int(1).to_bytes(1, "big"),
+    "focus_fine": int(2).to_bytes(1, "big"),
+    "focus_coarse": int(3).to_bytes(1, "big")
+}
+print(1)
```
Comment (HPrickettMorgan) [inline stepper_controller.py]: Clean up extra prints.

File: api.py (hunk 1 above)
Comment (jazima) [reply api.py:169]: 200 is a full rotation on the stepper so I was thinking this would never happen, I just included it to cover the edge case.

Comment (HPrickettMorgan) [reply api.py:169]: ah cool, sounds good then.

**Pull request merged**